Motor Movement Script

GrubeCube opened the pull request:
Motor movement script given by Gabriel. Feel free to move the review to someone else.
@nbice

Foo-Guy committed:
Add files via upload

Close to, if not the, final

GrubeCube committed:
Remove previous motor movement script

Previous motor movement script is no longer necessary in
favor of new one (MotorMove.py). Further modification
will need to be done to integrate all whell movement types.

nb-school commented on MotorMove.py:
```diff
@@ -0,0 +1,29 @@
+
+
+if __name__ == "__main__":
+    M = Motor(10, 11)
```
We need a way to limit speed. There is a min and max that is allowed to be set right? For instance we can send 1000 because that will exceed the speed that the motor can move. We should also have constants for all of the pins above.

GrubeCube replied:
Commit 7fed4e32e269cf32748a9d706272f3b21858a4c6 changed the way we set speed by only allowing the MovementManager to set the frequency. A max and min should not be necessary now.

nb-school commented on MotorMove.py:
```diff
@@ -0,0 +1,29 @@
+if __name__ == "__main__":
+    M = Motor(10, 11)
+    M.SetSpeed(130)
+    print('Hello')
```
'Hello'? We can get rid off this.

nb-school commented on MotorMove.py:
```diff
@@ -0,0 +1,29 @@
+
+if __name__ == "__main__":
+    M = Motor(10, 11)
+    M.SetSpeed(130)
```
Is this an arbitrary value? If not we can make a constant above.

GrubeCube replied:
Commit 7fed4e32e269cf32748a9d706272f3b21858a4c6 changed default values to be stored in constants instead of using the value inline.

nb-school commented on MotorMove.py:
```diff
@@ -0,0 +1,29 @@
+        dirSet = io.HIGH if speed < 0 else io.LOW
+        speedAbs = abs(speed)
+        self.motor.ChangeFrequency(speedAbs)
+        self.motor.start(50)
```
What is the value 50 here?

GrubeCube replied:
The value is the duty cycle, or the amount of time per period the motor should be working. Commit 7fed4e32e269cf32748a9d706272f3b21858a4c6 changed this value to use a constant instead of the inline value.

nb-school reviewed (changes requested):
We will need to set this up for 4 motors or pass in pin numbers. I suggest having all 4 motors on the same script and having a different method for each direction. For instance the froward direction would move the two left motors at the same speed and the two right motors at the same opposite speed (rotating in the opposite direction).

GrubeCube committed:
Add movement manager for ease of movement

Previously, only a single motor was programed to be able to run.
If full UGV movement was desired, the invoking program had to
manually set each motor. Movement manager simplifies this by
adding wrapper functions for standard movement types.

GrubeCube committed:
Switch motor movement logic and small format changes

The first interation of motor movement logic was incorrect as the
orientation of the motors was not taken into account. This commit
should fix the logic to make sense based on orientation. The
direction of each motor was guessed and might need to be flipped
later.

Formatting changes including putting two spaces between methods
and adding an author header.
